modules/calc/src/main/java/com/opengamma/strata/calc/runner/function/CalculationFunction.java

```diff
@@ -44,6 +46,15 @@
 public interface CalculationFunction<T extends CalculationTarget> {
 
   /**
+   * Gets the target type that this function applies to.
+   * <p>
+   * The target type will be matched exactly, thus it should be a class, not an interface.
```

Comment:
Is this a wise restriction? Dealing with subtyping might be messy but Sesame had this same restriction and it was a source of problems.

It's probably not something to tackle now but we should think about it and probably leave the door open to it just in case.